flow-report/src/common.tsx

```diff
@@ -7,25 +7,62 @@
 import {FunctionComponent} from 'preact';
 
 import {NavigationIcon, SnapshotIcon, TimespanIcon} from './icons';
+import {getScreenDimensions, getScreenshot} from './util';
 
 export const Separator: FunctionComponent = () => {
   return <div className="Separator" role="separator"></div>;
 };
 
+export const FlowStepIcon: FunctionComponent<{mode: LH.Result.GatherMode}> = ({mode}) => {
+  return <>
+    {
+      mode === 'navigation' && <NavigationIcon/>
+    }
+    {
+      mode === 'timespan' && <TimespanIcon/>
+    }
+    {
+      mode === 'snapshot' && <SnapshotIcon/>
+    }
+  </>;
+};
+
 export const FlowSegment: FunctionComponent<{mode?: LH.Result.GatherMode}> = ({mode}) => {
   return (
     <div className="FlowSegment">
       <div className="FlowSegment__top-line"/>
       {
-        mode === 'navigation' && <NavigationIcon/>
-      }
-      {
-        mode === 'timespan' && <TimespanIcon/>
-      }
-      {
-        mode === 'snapshot' && <SnapshotIcon/>
+        mode && <FlowStepIcon mode={mode}/>
       }
       <div className="FlowSegment__bottom-line"/>
     </div>
   );
 };
+
+export const FlowStepThumbnail: FunctionComponent<{
+  reportResult: LH.ReportResult,
+  width?: number,
+  height?: number,
+}> = ({reportResult, width, height}) => {
+  const screenshot = getScreenshot(reportResult);
+
+  // Crop the displayed image to the viewport dimensions.
+  const dimensions = getScreenDimensions(reportResult);
+  if (width && height === undefined) {
+    height = dimensions.height * width / dimensions.width;
+  } else if (height && width === undefined) {
+    width = dimensions.width * height / dimensions.height;
+  }
+
+  return <>
+    {
+      screenshot &&
+        <img
+          className="FlowStepThumbnail"
+          data-testid="FlowStepThumbnail"
```

Comment:
First time noticing `data-testid`. https://testing-library.com/docs/queries/bytestid/

What's the purpose of setting "testid" if it's just the same value as "className"? What are your thoughts on just using `root.container.querySelector`?

Otherwise, should we be setting a11y attributes to gel with the "guiding principles": https://testing-library.com/docs/queries/about#priority (this is the same reason puppeteer is moving towards accessibility selectors)

Reply:
I started using it because `getByTestId` throws automatically if it's not found and returns a DOM preview. `root.container.querySelector` just returns `null`.

You're right that this one probably doesn't need it. `getByRole('img')` should work fine.